samples/cpp/OpenCVsamples_cpp.h.in

```diff
@@ -0,0 +1,2 @@
+// the configured options and settings for Tutorial
+#define PATH_DATA_SAMPLE_OPENCV "@PROJECT_SOURCE_DIR@/samples/data"
```

Comment:
I understand your answer. I have check mailing list and I don't find people having problem with this relative path. I work in configuration 3 and with this relative path I have to change source file or duplicate data folder on three computer for different configuration of opencv. I think it was a good idea for everybody. I think now it's a good idea for configuration (3). People using (3) can change themself source file or data path like me.
I think I can make (4) but it means that prepocessor must be included in sample source files. In  my opinion  source sample files must be short to be readable. 
If you are agree I can close this request



PS sorry for my bad english